_posts/2016-09-21-approaching-tdd-outside-android.md

```diff
@@ -0,0 +1,159 @@
+
+The problem description is as follows:
+
+Create a simple bank application with the following features:
+
+<ul>
+<li>Deposit into Account</li>
+<li>Withdraw from Account</li>
+<li>Show a bank statement</li>
+</ul>
+
+The original kata provides a class with the following structure:
+
+````
+public class Account {
+    public void deposit(int amount) {
+    }
+
+    public void withdraw(int amount) {
+    }
+
+    public void showStatement() {
+    }
+}
+````
+
+And one constraint: You are not allowed to add any other public method to this class.
+We have now used the outside-in TDD workflow in numerous projects and have adapted some aspects of it to suit developing on Android specifically. To demonstrate this, and to make it easier to follow, this post will follow the "Bank Kata" that Sandro Mancuso uses on his 'Outside-In' screencast. (You don't have to watched this screencast to follow along but it's useful as a primer on the concepts.) In his screencast the problem isn't build for Android, but we found it useful to have it as a focus.
+
+## But, we broke that rule
+
+It's worth mentioning that we broke this constraint somewhat to add another public method to the Account class. We did this to in order to attach the view to the account object. We needed to do so because we’ve chosen to use Android activities, and as we all know and suffer, they are instantiated by the system. Therefore, we can not pass the view through the BankAccount class constructor.
+
+We could have avoided adding this method by creating some kind of presenter, using the Account object there instead of calling it from the Activity. We consider it would make the code more complex and as it’s just a kata we’ve decided not to do so.
+
+## Extracting the acceptance criteria
+Let's start by describing the best-case workflow for tackling this problem. The starting point should ideally be a user story defining what needs to be done, who are we building it for, and why we are building it. For further information about how to write users stories, visit this <a href="https://sprint.ly/blog/agile-user-stories/?utm_content=buffer2dda2">link</a>
+
+The starting point should be a <a href="http://dannorth.net/whats-in-a-story/">user story</a> defining: What needs to be done, who are we building it for and why we are building it.
+
+The Bank Kata problem description above talks about three features so there should be three user stories. For the purpose of this post, we are going to focus on the show statement feature. The user story for the show statement could be written as follows:
```

Comment:
I would not necessary agree that three features means three user stories. Is the first sentence necessary at all?